components/pedigree/resources/src/main/resources/pedigree/pedigreeDate.js

```diff
@@ -200,11 +209,15 @@ define([], function(){
         },
 
         // Returns an integer or null.
-        // Iff "failsafe" returns first year of the decade if year is not set and decade is
-        getYear: function(failsafe) {
+        // Iff "failsafe" returns a value even if only decade is availabe (and year is not):
+        //  - iff "average" the middle year of the decade is returned, otherwise the first year of the decade.
+        getYear: function(failsafe, average) {
             if (this.isComplete() && this.year == null && failsafe) {
                 // remove trailing "s" from the decade && convert to integer
                 var year = parseInt( this.decade.slice(0,-1) );
+                if (average) {
+                    year += 5;
```

Comment:
What happens if the date is even fuzzier, like "1700s"? Will this block never be reached, since `isComplete` returns false?